### PR TITLE
SA-2318 Automate Changelog on SDK Release

### PR DESCRIPTION
**Pull request by jworkmanjc**
## Issues
* [SA-2318](https://jumpcloud.atlassian.net/browse/SA-2318) - Automate Changelog on SDK release

## What does this solve?

There should be a changelog when we release new versions of our SDKs. This PR addresses that. 
In addition a process is introduced to the CI workflow to automatically update the git diff of the files changed in the sdk's /generated/custom directory. 

Lastly on merge to main branch the CI process will automatically draft a release of each SDK and add the corresponding version notes from the changelog file. Each changelog file is located in the root of the repo ex. JumpCloud.SDK.V1.md

## Is there anything particularly tricky?

This is PR is not intended to result in a SDK release. 

## How should this be tested?

The Build-SDKChangelog function can be used to test the functionality of this PR. 

Test 1)
Update the PSD1 version of an SDK function & from the root of the project run `Tools/BuildSDKChangelog -SDKName JumpCloud.SDK.V1` (or which ever SDK you changed).
The ChangeLog File for the corresponding SDK should be updated with a new version block containing the version you've specified in the PSD1 File.

Ex. Modify /SDKs/PowerShell/JumpCloud.SDK.V1/JumpCloud.SDK.V1.psd1 module version to 2.0.0
Run `Tools/BuildSDKChangelog -SDKName JumpCloud.SDK.V1` from the root of the project
'./JumpCloud.SDK.V1.md' should have been updated with a 2.0.0 version header

Test 2)
After running test 1, fill out some data in the {{ Replace me blocks }} and run `Build-SDKChangelog -SDKName JumpCloud.SDK.V1` again, nothing should have changed in the changelog file.

Test 3)
After running test 2:
locally add any .ps1 file to /SDKs/PowerShell/JumpCloud.SDK.V1/custom/generated
locally modify any existing .ps1 file to /SDKs/PowerShell/JumpCloud.SDK.V1/custom/generated
locally remove any existing .ps1 file in /SDKs/PowerShell/JumpCloud.SDK.V1/custom/generated

locally commit changes but **do not** push to remote.

Run `./Tools/Build-SDKChangelog -SDKName JumpCloud.SDK.V1` The autogenerated section should update and display your added, modified & removed file in the "Functions Added, Modified & Removed" blocks

## Screenshots
Changelog File:
Screen Shot 2022-03-25 at 8 22 25 AM
On SDK merge to main, a draft release is autogenerated as a CI task:
Screen Shot 2022-03-25 at 8 25 29 AM

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -70,15 +74,45 @@ build-sdks: &build-sdks
...
         - Build and Test JumpCloud SDK V2
+        - Draft JumpCloud SDK V2 GitHub Release
         - Commit Changes
         - Pack and Deploy To CodeArtifact
         - PSGallery Approval
       filters:
         branches:
           only:
             - master
+  - Draft JumpCloud SDK DirectoryInsights GitHub Release:
```
Comment (jworkmanjc) [inline .circleci/config.yml:89]: Each SDK release is drafted at this stage

File: ApiTransform.ps1 [hunk 2]
```diff
@@ -404,7 +425,33 @@ $TransformConfig = [Ordered]@{
...
+            '/providers/{provider_id}/integrations',
+            '/providers/{provider_id}/integrations/connectwise',
+            '/integrations/connectwise/{UUID}',
+            '/integrations/connectwise/{UUID}/settings',
+            '/integrations/connectwise/{UUID}/agreements',
+            '/integrations/connectwise/{UUID}/agreements/{agreement_ID}/additions',
+            '/integrations/connectwise/{UUID}/companies',
+            '/integrations/connectwise/{UUID}/companytypes',
+            '/integrations/connectwise/{UUID}/mappings',
+            # '/providers/{provider_id}/administrators',
```
Comment (jworkmanjc) [inline ApiTransform.ps1:440]: Remove uncommented functions here, these will be implemented in SA-2319

File: Tools/Build-SdkChangelog.ps1 [hunk 3]
```diff
@@ -0,0 +1,88 @@
+[CmdletBinding()]
+param (
+    [Parameter(Mandatory = $true, ValueFromPipelineByPropertyName = $true, HelpMessage = 'Name of the API to build an SDK for.')][ValidateSet('JumpCloud.SDK.DirectoryInsights', 'JumpCloud.SDK.V1', 'JumpCloud.SDK.V2')][ValidateNotNullOrEmpty()][System.String[]]$SDKName
+)
+$SdkChangelogFilePath = "./$SDKName.md"
+Import-Module ($PSScriptRoot + '/New-SdkChangelog.ps1')
+# For authorized requests add header & OAuth Token:
+# $headers = @{
+#     "Authorization" = "<OAuth Token>"
+# }
```
Comment (jworkmanjc) [inline Tools/Build-SdkChangelog.ps1:10]: Rate limit occurs when this triggers many times in one hour, we can add in Authorization via OAuth Tokens if this becomes an issue

**Review by gweinjc — APPROVED** [review]
Tested locally and appears to be working as intended

File: ApiTransform.ps1 (hunk 2 above)
Comment (jworkmanjc) [reply ApiTransform.ps1:440]: Will resolve in next PR to main

**Pull request merged**